pkg/prometheus/promcfg.go

```diff
@@ -348,6 +381,46 @@ func addTLStoYaml(cfg yaml.MapSlice, namespace string, tls *monitoringv1.TLSConf
 	return cfg
 }
 
+func (cg *ConfigGenerator) MergeTLSConfigWithScrapeClass(tlsConfig *monitoringv1.TLSConfig, scrapeClass *monitoringv1.ScrapeClass) *monitoringv1.TLSConfig {
```

Comment:
Could we have a unit test for this function?

I'm afraid this will become quite big as we add more fields to ScrapeClass in the future, a test will help us keep this working :)